qiita_db/test/test_metadata_template.py

```diff
@@ -652,6 +775,133 @@ def test_create(self):
                ['2.Sample3', "Value for sample 3"]]
         self.assertEqual(obs, exp)
 
+    def test_create_int_prefix(self):
+        """Creates a new SampleTemplate"""
+        st = SampleTemplate.create(self.metadata_int_pref, self.new_study)
+        # The returned object has the correct id
+        self.assertEqual(st.id, 2)
+
+        # The relevant rows to required_sample_info have been added.
+        obs = self.conn_handler.execute_fetchall(
+            "SELECT * FROM qiita.required_sample_info WHERE study_id=2")
+        # sample_id study_id physical_location has_physical_specimen
+        # has_extracted_data sample_type required_sample_info_status_id
+        # collection_timestamp host_subject_id description
+        exp = [["2.12.Sample1", 2, "location1", True, True, "type1", 1,
```

Comment:
Are the study ids aggregated to the sample ids used anywhere in the code? AFAIK, there is no place where such study  id is used, and it is aggregated at the upload time. The only reason why they are added is because it is an easy way to define a DB global sample id...